tests/splinterdb_test_apis.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * -----------------------------------------------------------------------------
+ * splinterdb_test_apis.h --
+ *
+ * Header file with extern prototypes for testing-hooks provided in diff source
+ * files.
+ * -----------------------------------------------------------------------------
+ */
+#pragma once
+
+#include "splinterdb/splinterdb.h"
+
+// External APIs provided -ONLY- for use as a testing hook.
+void
+splinterdb_cache_flush(const splinterdb *kvs);
+
+platform_heap_id
+splinterdb_get_heap_id(const splinterdb *kvs);
+
+const void *
```

Comment:
Move this file to `src/splinterdb_private.h` and `#include` it in `splinterdb.c` so that the decls get type-checked against the definitions.

Reply:
Done, with one minor diff. I wanted to retain the fact that this `.h` exists solely for use by tests, so renamed this file to `src/splinterdb_tests_private.h`.